securedrop_client/queue.py

```diff
@@ -41,10 +41,15 @@ def _process(self, exit_loop: bool) -> None:
                 job._do_call_api(self.api_client, session)
             except RequestTimeoutError:
                 self.last_job = job  # "cache" the last job since we can't re-queue it
-                return
+            except ApiInaccessibleError:
+                # This is a guard against #397, we should pause the queue execution when this
+                # happens in the future and flag the situation to the user (see ticket #379).
+                logger.error('Client is not authenticated, skipping job...')
+            finally:
+                # process events to allow this thread to handle incoming signals
+                QApplication.processEvents()
```

Comment:
Looks like the only incoming signals this thread needs to handle are the download_file_thread.started and main_thread.started signals, which only start this processing loop, right? When the loop is running, there shouldn't need to be any handling of incoming signals right? I'm curious if this really needs to be called. The docs say:

> The event loop is started with a call to exec(). Long-running operations can call processEvents() to keep the application responsive.

But we're already in a separate thread from the application, so we don't need to worry about the application not being able to respond, _I think_.

Reply:
so i thought that this is needed for processing events from queued signal/slot connections - let's file a followup for this - if that's wrong then this is important to understand in more detail (i'm just moving it here for readability in this diff and want to avoid tacking on more changes here)